### DISCUSSION TITLE
[@types/jsdom] pulling in dom.iterable has side-effects

### DISCUSSION THREAD

**Discussion by hendriksollich** [DefinitelyTyped/DefinitelyTyped#63157] [Issues with a @types package]
After [updating](https://app.renovatebot.com/package-diff?name=@types%2fjsdom&from=20.0.0&to=20.0.1) `@types/jsdom` to `20.0.1` if you have an object that implements e.g. `RTCStatsReport` you suddenly also have to implement `ReadonlyMap<string, any>`, since that is being pulled in with `lib.dom.iterable.d.ts`.

This is a bit of a weird side-effect of a minor types update. Can this be avoided somehow?

Thanks

[a comment by an automated account is not shown]

**Comment by haoqunjiang** [comment]
I'm sorry for the inconvenience my PR has caused.

However, considering WebRTC was never supported in `jsdom`[^1], it's not the intended use case for these related global interfaces that `@types/jsdom` accidentally brought in to be relied upon by users.

So, in my opinion, if one needs to implement `RTCStatsReport`, the correct way is to declare their own version of this interface, and thus won't be affected by this update.

[^1]: https://github.com/jsdom/jsdom/issues/1746

**Reply by hendriksollich** [reply]
The fact that WebRTC is affected is more incidental. My question is rather, should a library make use of these [`+ /// <reference lib="..." />`](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-lib-) directive, since they seem to have global side-effects. In my case, the affected package is not even using jsdom at all, it just shares the same mono-repo. 
In the docs it sais 
> For example, adding /// <reference lib="es2017.string" /> to one of the files in a compilation is equivalent to compiling with --lib es2017.string.

Do other @types/... packages do this?
